### PR TITLE
Add the execution system ID to apps endpoints.

### PR DESCRIPTION
**Pull request by dennisroberts71**
The current way of routing calls to different execution systems (e.g. Agave, vs the DE's HTCondor cluster) is brittle because it relies on the fact that DE app IDs are UUIDs whereas Agave's are not. This change adds the system ID to the URI path so that calls can be routed in a much more extensible manner.

This change is a work in progress. Code that has ben aded for endpoints that haven't been converted yet is subject to change further changes.

### CODE REVIEW

File: src/apps/service/apps/agave.clj [hunk 1]
```diff
@@ -68,33 +74,60 @@
...
+
   (formatPipelineTasks [_ pipeline]
     (pipelines/format-pipeline-tasks agave pipeline))
 
   (listJobs [self params]
     (job-listings/list-jobs self user params))
 
+  ;; TODO: Determine how this should be refactored now that we have system IDs. If I remember correctly,
+  ;; this method is used during job listings. If that's the case, we can filter apps by execution system
+  ;; easily enough.
```
Comment (strootman) [inline src/apps/service/apps/agave.clj]: If there is not a ticket for this, there should be.

File: src/apps/service/apps/agave.clj [hunk 2]
```diff
@@ -105,6 +138,10 @@
     (when-not (util/uuid? (:app_id submission))
       (agave-jobs/submit agave user submission)))
 
+  (submitJob [this system-id submission]
+    (validate-system-id system-id)
+    (agave-jobs/submit agave user submission))
+
```
Comment (strootman) [inline src/apps/service/apps/agave.clj]: Not having to check that the `app_id` is _not_ a UUID is a good change!

File: src/apps/service/apps/agave.clj [hunk 3]
```diff
@@ -139,61 +180,106 @@
...
+    (validate-system-id system-id)
+    (reject-app-documentation-edit-request))
+
   (adminAddAppDocs [_ app-id _]
     (when-not (util/uuid? app-id)
       (reject-app-documentation-edit-request)))
 
+  (adminAddAppDocs [_ system-id app-id _]
+    (validate-system-id system-id)
+    (reject-app-documentation-edit-request))
```
Comment (strootman) [inline src/apps/service/apps/agave.clj]: Are there service tests for these endpoints which return `bad requests`? Seems like easy pickings.

**Comment by strootman** [comment]
This PR looks LGTM. The corresponding ticket may be difficult to validate in its current state. I think it is necessary to list every endpoint affected/added. Also, the endpoints will need to be added to the drop notes as well.

File: src/apps/service/apps/agave.clj (hunk 1 above)
Comment (dennisroberts71) [inline src/apps/service/apps/agave.clj]: I was going to do this under the same ticket. The comment is just a reminder for myself.

File: src/apps/service/apps/agave.clj (hunk 3 above)
Comment (dennisroberts71) [inline src/apps/service/apps/agave.clj]: There aren't yet, but there will be once the admin endpoints are converted.

**Comment by dennisroberts71** [comment]
Thanks for the review! I'll create subtasks for each of the individual repositories for extra Jirosis.

**Comment by strootman** [comment]
We discussed this offline, but adding it here.
I didn't mean to add one jira per endpoint, just that the endpoints be listed in CORE-8065 (or where appropriate). This was only to facilitate the ability to properly validate the ticket/work.

**Comment by strootman** [comment]
Last thing. Be sure to rebase your changes to the master branch and force push your branch before merging this PR.

**Pull request merged**